homeassistant/components/hunterdouglas_powerview/select.py

```diff
@@ -67,26 +62,22 @@ async def async_setup_entry(
 
     pv_entry: PowerviewEntryData = hass.data[DOMAIN][entry.entry_id]
 
-    entities = []
-    for raw_shade in pv_entry.shade_data.values():
-        shade: BaseShade = PvShade(raw_shade, pv_entry.api)
-        if SHADE_BATTERY_LEVEL not in shade.raw_data:
-            continue
-        name_before_refresh = shade.name
-        room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
-        room_name = pv_entry.room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
-
-        for description in DROPDOWNS:
-            entities.append(
-                PowerViewSelect(
-                    pv_entry.coordinator,
-                    pv_entry.device_info,
-                    room_name,
-                    shade,
-                    name_before_refresh,
-                    description,
-                )
-            )
+    entities: list[PowerViewSelect] = []
+    for shade in pv_entry.shade_data.values():
+        if shade.has_battery_info() is True:
```

Comment:
```suggestion
        if not shade.has_battery_info():
            continue
```

Please reverse conditionals and continue to avoid long indents